lib/components/serverside-data/ServersideTable.vue

```diff
@@ -14,26 +31,41 @@
 import omit from 'lodash/omit'
 import ServersideIterator from './ServersideIterator.vue'
 import Table from './Table.vue'
+import CustomTable from './CustomTable.vue'
+
+const stylesEnum = {
+  table: 'table',
+  customTable: 'custom-table',
+}
 
 export default {
   name: 'ServersideTable',
   components: {
     ServersideIterator,
     Table,
+    CustomTable,
   },
   inheritAttrs: false,
   props: {
     icon: {
       type: String,
       default: () => '',
     },
+    tableStyle: {
+      type: String,
+      default: () => stylesEnum.table,
+      validator: (v) => Object.values(stylesEnum).includes(v),
+    },
   },
   data() {
     return {
       loading: true,
     }
   },
   computed: {
+    showCustomTable() {
+      return this.tableStyle === stylesEnum.customTable
+    },
```

Comment:
We discussed not using a boolean in order to support more table styles in the future. Now you use the boolean again but in a different spot 😄 . You either can do:
``` javascript
<CustomTable v-if="tableStyle === stylesEnum.customTable">
<Table v-else-if="tableStyle === stylesEnum.table">
```
Or you define a mapping between the table style and the component and use a `<component :is`:
```
const tableStyles = {
  table: 'Table',
  customTable: 'CustomTable',
}
```
This way add a new style is very simple.